### ISSUE TITLE
Fix Functions on App Service Plan logic

### ISSUE THREAD

**Issue by m-mertz** [Azure/azure-libraries-for-net#485]
It feels like the Functions logic in its current form was written for the consumption plan model, and app service plan support was tacked on without proper testing. There are a few issues pointing at specific symptoms of this, but no "this needs an overhaul" type issue, so here goes.

The scenario I’d like to see enabled:

```csharp
IFunctionApp app = await appServiceManager.FunctionApps
    .Define(appName)
    .WithExistingAppServicePlan(appServicePlan)
    .WithExistingResourceGroup(resourceGroup)
    .WithExistingStorageAccount(storageAccount)
    .WithRuntimeVersion("~2")
    .CreateAsync()
    .ConfigureAwait(false);
```

Expected outcome: A new function app with the same app settings as a new app service plan function app created in the portal, and no weird side effects.

Gaps:

1. SDK creates WEBSITE_CONTENTAZUREFILECONNECTIONSTRING app setting.
2. SDK creates WEBSITE_CONTENTSHARE app setting.
3. SDK creates file share in the specified storage account (the file share referenced in setting above). I don’t just want the setting to disappear, I’d also like to avoid creating garbage data in the storage account.
4. When retrieving the function app again in the SDK, it should recognize the configured storage account by AzureWebJobsStorage setting alone (if running on an app service plan).

The mentioned settings are for consumption plans only: https://docs.microsoft.com/en-us/azure/azure-functions/functions-app-settings#websitecontentazurefileconnectionstring

Related issues: #457, #265

**Comment by xenalite** [comment]
@hovsepm @jianghaolu I just started using fluent APIs for Function Apps and hit this exact issue and the ones linked in related. One of the original issues has been open for over a year. Are there any plans to resolve these?
I attempted to get around this problem by provisioning the function app using a generic ResourceManager, but I cannot specify `Kind` property to be `functionApp`.

**Comment by yaohaizh** [comment]
Fixed by PR#761.
